### ISSUE TITLE
dropbox broken due to hardcoded /usr/share/X11

### ISSUE THREAD

**Issue by anderspapitto** [NixOS/nixpkgs#16744]
## Issue description

dropbox runs but won't accept any keyboard input (which makes it impossible to log in). At startup, it also gives an error message about /usr/share/X11

### Steps to reproduce

```
$ /nix/store/0cdrakmdfgycm59kvwyljbfpmfvrrdyx-dropbox-4.4.29/bin/dropbox 
xkbcommon: ERROR: failed to add default include path /usr/share/X11/xkb
```

### Diagnosis

I've done some digging, and found a couple things. First, dropbox runs fine when /usr/share/X11 is bind mounted to the correct location in the nix store.

```
[anders@gurney:~]
$ unshare -m -r
[root@gurney:~]
$ mount --bind /nix/store/shi5k1kl27dz1ka1ngriqq1spqjza70l-xorg-server-1.18.3/ /usr/
[root@gurney:~]
$ /nix/store/0cdrakmdfgycm59kvwyljbfpmfvrrdyx-dropbox-4.4.29/bin/dropbox
```

also, the hardcoded string appears to be coming from `libQt5XcbQpa.so`, which possibly includes a bundled copy of libxkbcommon.

```
[anders@gurney:~]
$ strings /nix/store/0cdrakmdfgycm59kvwyljbfpmfvrrdyx-dropbox-4.4.29/opt/dropbox/libQt5XcbQpa.so.5 | grep /usr/share
/usr/share/X11/xkb
/usr/local/share/:/usr/share/
/usr/share/pixmaps
```

## Technical details

```
[anders@gurney:~]
$ nixos-version
16.09.git.c7000ef (Flounder)
[anders@gurney:~]
$ nix-env --version
nix-env (Nix) 1.11.2
[anders@gurney:~]
$ nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion
"16.09.git.c7000ef"
```

**Comment by FRidh** [comment]
cc maintainer @ttuegel

**Comment by ttuegel** [comment]
Thanks for reporting! It looks like we'll have to replace the bundled Qt with our own build. Not a big deal, we've done that before. I'll have this fixed as soon as I figure out what Qt version Dropbox is using.

**Comment by ttuegel** [comment]
What revision are you on? This was already fixed in 18cc8d482dab52e7833c69e83f484e4b14218a1a.

**Comment by mikoim** [comment]
Sry, I used old revision aeff3080d02ce6ac2b2ad0b0029c1388eab2d8c7 via nixos-unstable binary channel. Building from HEAD and it works perfectly. Thank you for your package maintaining.
